### PR TITLE
Implement PoC version of Pipeable.Script

### PR DESCRIPTION
**Pull request by ibratoev**
Add Script.swift as a proof of concept for how the scripting logic would be implemented. 
Add ScriptTests.swift for testing Script functionality.

Implement async calls counting so that a script run can wait for all pending completions to finish.

### CODE REVIEW

File: PipeableTests/ScriptTests.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@testable import Pipeable
+import XCTest
+
+final class ScriptTests: XCTestCase {
+    func test_goto() throws {
+        let result = runScript("""
```
Comment (pdobrev) [inline PipeableTests/ScriptTests.swift:6]: How does this work in Swift if it's not async? Does it just block the calling thread? 

Not sure what we want here in terms of API -- sync or async? I was defaulting to async, but wondering if you have other thoughts.

Comment (ibratoev) [reply PipeableTests/ScriptTests.swift:6]: Currently, it blocks the main thread as it is sync API. I should be `async`, I will get that fixed before merging the PR.

**Review by pdobrev — APPROVED** [review]
If i understand correctly, it seems that we'll need to wrap the `Pipeable` Swift public API in calls `withCompletion` once and then another wrapper that goes from callbacks to promises to expose to the scripts.

I wonder if down the line it makes sense to somehow automate this (even though code gen), since it seems like as the API surface grows, it may be a source of unintended errors (just cause it's a lot of copy-paste essentially).

This is great for poc, I think for scripts we should probably standardize on some common format like an exported async function that takes a page or something, but we can figure this out as we formalize it.

**Comment by pdobrev** [comment]
@ibratoev In light of #11 where we'll need to do a bunch of renaming and reorganization (to publish the framework as `PipeableSDK`), I hope you don't mind if I go ahead and merge this PR. It will reduce nasty merge conflicts and since this is POC, it will likely have further evolutions as we get it closer to production.

**Pull request merged**